sql/core/src/test/scala/org/apache/spark/sql/execution/adaptive/AdaptiveQueryExecSuite.scala

```diff
@@ -554,7 +554,7 @@ class AdaptiveQueryExecSuite
       val smj = findTopLevelSortMergeJoin(plan)
       assert(smj.size == 2)
       val smj2 = findTopLevelSortMergeJoin(adaptivePlan)
-      assert(smj2.size == 2, origPlan.toString)
+      assert(smj2.size == 1, origPlan.toString)
```

Comment:
This is not the right fix, and I am working on this to see if I can make the cost higher. For SMJ -> BHJ, it may not be possible with this PR.

Reply:
@cloud-fan/@maryannxue Do you have a suggestion on an example where a plan change could introduce a greater cost? I haven't been able to come up with one for the SMJ -> BMJ scenario. Thanks in advance!

Reply:
just make an outer join so your change won't apply?

Reply:
Simply changing it to outer join may not work. For example,
```
SELECT * FROM t1 LEFT JOIN t2 ON t1.a = t2.c LEFT JOIN t2 as t3 ON t2.c = t3.c
```
For the left outer join between `t1` and `t2`, you can only build right side (`t2`), but the resulting output partitioning is from this join is on the left side (`t1`). Thus, the join between `t2` and `t3` will always introduce shuffle and this will not help getting the higher cost.

(On top of this, putting any `WHERE` clause would convert outer join to inner join. :))

Reply:
@cloud-fan (the initial tagging didn't work), do you have a suggestion on an example where a plan change could introduce a greater cost?